lms/djangoapps/program_enrollments/rest_api/v1/tests/test_views.py

```diff
@@ -1880,8 +1888,11 @@ def test_due_dates(self):
             ]
             due_dates = response.data['course_runs'][0]['due_dates']
 
-            for block in block_data:
-                self.assertIn(block, due_dates)
+            if course_in_progress:
+                for block in block_data:
+                    self.assertIn(block, due_dates)
```

Comment:
nit-level suggestion:
```suggestion
                assert set(block_data) <= set(due_dates)
```
might make the test output more readable; I haven't tried, though.

Reply:
yeah this doesn't work since it would try create a set of dicts